decorator.rb

```diff
@@ -0,0 +1,25 @@
+require_relative 'nameable'
+
+class Decorator
+  include Nameable
```

Comment:
- [ ] Same here kindly make sure `Decorator` inherit from `Nameable` instead

```suggestion
class Decorator < Nameable
```